include/stl2/view/single_ref.hpp

```diff
@@ -0,0 +1,74 @@
+// cmcstl2 - A concept-enabled C++ standard library
+//
+//  Copyright Eric Niebler 2017
+//  Copyright Dalton M. Woodard 2019
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/caseycarter/cmcstl2
+//
+#ifndef STL2_VIEW_SINGLE_REF_HPP
+#define STL2_VIEW_SINGLE_REF_HPP
+
+#include <memory>
+#include <utility>
+
+#include <stl2/type_traits.hpp>
+#include <stl2/detail/fwd.hpp>
+#include <stl2/detail/concepts/object.hpp>
+#include <stl2/detail/view/view_closure.hpp>
+#include <stl2/view/view_interface.hpp>
+
+STL2_OPEN_NAMESPACE {
+namespace ext {
+	template<class T>
+	requires std::is_object_v<T>
+	class single_ref_view : public view_interface<single_ref_view<T>> {
```

Comment:
s/`class`/`struct`/

For future reference, in cmcstl2:
1. `class` means template parameter
2. `struct` means user-defined type
3. `typename` means dependent type